c2corg_api/views/navitia.py

```diff
@@ -105,3 +140,787 @@ def get(self):
+
+    src_epsg = 3857
+    transformer = Transformer.from_crs(
+        f"EPSG:{src_epsg}", "EPSG:4326", always_xy=True)
+    lon, lat = transformer.transform(geom.x, geom.y)
+
+    journey_params['to'] = f"{lon};{lat}"
+
+    destination_coverage = get_coverage(lon, lat)
+
+    try:
+        # Récupération de la clé API depuis les variables d'environnement
+        api_key = os.getenv('NAVITIA_API_KEY')
+        if not api_key:
+            raise HTTPInternalServerError(
+                'Configuration API Navitia manquante')
+
+        response = {}
+
+        if (destination_coverage):
+            # Appel à l'API Navitia Journey with coverage
+            response = requests.get(
+                f'https://api.navitia.io/v1/coverage/{destination_coverage}/journeys',  # noqa: E501
+                params=journey_params,
+                headers={'Authorization': api_key},
+                timeout=30
+            )
+        else:
+            # Appel à l'API Navitia Journey
+            response = requests.get(
+                'https://api.navitia.io/v1/journeys',
+                params=journey_params,
+                headers={'Authorization': api_key},
+                timeout=30
+            )
+
+        # Vérification du statut de la réponse
+        if response.status_code == 401:
+            raise HTTPInternalServerError('Authentication error with Navitia API')  # noqa
+        elif response.status_code == 400:
+            raise HTTPBadRequest('Invalid parameters for Navitia API')
+        elif response.status_code == 404:
+            # no_destination -> public transport not reachable from destination
+            # no_origin -> public transport not reachable from origin
+            # these do not count as proper errors,
+            # more like the wp is just not reachable
+            if response.json()['error']['id'] != 'no_destination' and \
+               response.json()['error']['id'] != 'no_origin':
+                raise HTTPInternalServerError(response.json()['error'])
+            return False
+        elif not response.ok:
+            raise HTTPInternalServerError(f'Navitia API error: {response.status_code}')  # noqa: E501
+        else:
+            # code 200 OK
+            # make sure the waypoint is reachable if at least one journey's
+            # departure date time is the same day as the day in journey_params
+            for journey in response.json().get('journeys', []):
+                journey_day = int(journey['departure_date_time'][6:8])
+                param_day = int(journey_params['datetime'][6:8])
+                if journey_day == param_day:
+                    return True
+
+            return False
+
+    except requests.exceptions.Timeout:
+        raise HTTPInternalServerError(
+            'Timeout when calling the Navitia API')
+    except requests.exceptions.RequestException as e:
+        raise HTTPInternalServerError(f'{str(e)}')
```

Comment:
except requests.exceptions.Timeout:
    raise GatewayTimeout('Timeout when calling the Navitia API')

except requests.exceptions.ConnectionError:
    raise BadGateway('Connection error when calling the Navitia API')

except requests.exceptions.HTTPError as e:
raise BadGateway(f'Navitia API HTTP error: {e.response.status_code}')

except requests.exceptions.RequestException as e:
    raise BadGateway(f'Navitia API request error: {str(e)}')